ui-speedspacechart/src/components/helpers/drawElements/reticle.ts

```diff
@@ -127,6 +129,13 @@ export const drawCursor = (
         electricalModeText = `${isElectrified ? electrificationUsage.mode : electrificationUsage.object_type}`;
       }
 
+      const previousPowerRestriction = powerRestrictions.findLast(
+        (range) => range.start <= previousCurvePosition.position
```

Comment:
Shouldn't we also check `range.end` here? Or do we want to display text for the closest power restriction on the left even if we're past it?

Reply:
Nice catch, rectified the condition and change shortened one of the power restriction range so we can test this edge case easier.